### PR TITLE
New MPPI Cost Critic (Contrib: Brice Renaudeau)

### PR DESCRIPTION
**Pull request by SteveMacenski**
Uses cost values in costmap rather than obstacle distances for obstacle avoidance alternative option that some users have found better for their applications. 

@BriceRenaudeau @doisyg please review. Note parameter name / meaning (in one case) and critic name changes. The weight is now normalized in the code so that its in the general range as other parameter values. So whatever you use (0.015) multiply by 254 to get the new value (3.81). The name of weight was also changed to `cost_weight` to match the other critics, now that we don't have multiple weights to differentiate within the critic

https://github.com/ros-planning/navigation2/issues/4057

Docs PR: https://github.com/ros-planning/navigation.ros.org/pull/514

### CODE REVIEW

File: nav2_mppi_controller/include/nav2_mppi_controller/critics/cost_critic.hpp [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+  bool inCollision(float cost, float x, float y, float theta);
+
+  /**
+    * @brief cost at a robot pose
+    * @param x X of pose
+    * @param y Y of pose
+    * @param theta theta of pose
+    * @return Collision information at pose
+    */
+  float costAtPose(float x, float y);
```
Comment (BriceRenaudeau) [inline nav2_mppi_controller/include/nav2_mppi_controller/critics/cost_critic.hpp:68]: As said in the issue, the name is more "costAtPoint" as it does use the theta.
may need to change doxygen description

**Review by BriceRenaudeau — APPROVED** [review]
Seams ok for me, not tested.
Just a comment on the oxygen not fitting the function

**Comment by doisyg** [comment]
At first glance, works for me and very nice behavior improvement compared to the obstacle critics with default parameters. 
I did not test extensively across all potential corner cases but I gave it a proper sanity check run.

**Comment by SteveMacenski** [comment]
@BriceRenaudeau can you test it? Pending that we can merge

**Comment by BriceRenaudeau** [comment]
Hi Steeve, I will not be able to test it until Monday.

**Comment by doisyg** [comment]
Finally tested, works great with default parameters + `consider_footprint: true` + `regenerate_noises: true` + increased `wz_std` for my robot configuration.
Thanks a lot @BriceRenaudeau for proposing it. For non circular robots, this is a game changer.

**Comment by SteveMacenski** [comment]
Thanks for the review, testing, and development! It is sincerely appreciated!
